Sources/MockoloFramework/Models/ClassModel.swift

```diff
@@ -25,7 +25,7 @@ final class ClassModel: Model {
     let identifier: String
     let declType: DeclType
     let entities: [(String, Model)]
-    let initParamCandidates: [Model]
+    let initParamCandidates: [VariableModel]
```

Comment:
I think all of `Model` assigned to `initParamCandiates` are `VariableModel`. so type infromation should denote it.

Reply:
> Why you needs this change?

This is to improve the readability. `Model` is a bit unclear to imply what `initParamCandidates` means.
If we can treat `initParamCandidates` as `VariableModel` (or some specific Model Type), it might be helpful for others to understand what it is.

> Is there any reason that initParamCandidates should always be VariableModel?

In my understanding, all of candidates to use in initializer, so that mock class can capture init's parameter, are `VariableModel` and I couldn't come up with the counter-example.

[This block](https://github.com/uber/mockolo/blob/68eedb21a810986e936140ff4dc6662b14058148/Sources/MockoloFramework/Models/ParsedEntity.swift#L42-L58) returns only `VariableModel` because only `VariableModel.canBeInitParam` returns `true` and other Model returns `false`. [link](https://github.com/uber/mockolo/blob/68eedb21a810986e936140ff4dc6662b14058148/Sources/MockoloFramework/Models/Model.swift#L109-L111)

- example

```swift
/// @mockable
protocol MyProtocol {
    var param: Int { get } // this is initParamCandidate.
    init(param: Any)
    init(param: String)
    init(with param: [Character])
}
```



but this is kind of IMO. I need your opinion.🙏

Reply:
I didn't aware that `sortedInitVars` do filtering.
I also found that `initParamCandidates` is not a common one but a high context one.

Maybe there are few effects caused by changing the type of the property. OK.

Reply:
You can also remove `var canBeInitParam: Bool { get }` requirement in `Model`.